doc/adr/0011-tsp-analysis-api.md

```diff
@@ -178,100 +183,175 @@ sequenceDiagram
     client->>client: Refresh UI
 ```
 
-### Configuration service per experiment
+### Configure customizable outputs 
+
+Outputs might accept input parameters that will configure the analysis used to create the output. Such customization can create new outputs or provide additional query parameters. The TSP api will allow the back-end to define configuration types for that. Each configuration type will use `CustomizationTypeSource` to define what parameters need to be provided to create a new and derived output. The derived output descriptor will store the configuration details. Derived outputs can be deleted using a DELETE operation.
 
-For this data provider service will be augmented for managing configurations per experiment.
+    GET /experiments/{expUUID}/outputs/{outputId}/configTypes
+    Returns a list of configuration source types: typeId, name, description, scope, expected parameter descriptors (e.g. "path" for file path)
 
-    GET experiments/{expUUID}/outputs/config
-        returns a map typeId -> list of configuration descriptors of existing configurations on server
-    POST experiments/{expUUID}/outputs/config
-        Assign configuration to an experiment using typeId and configId from above.
-        Returns configuration descriptor and list of data provider descriptors (if available)
-    DELETE experiments/{expUUID}/outputs/config/{configId}
-        Removes a configuration from an experiment
+    GET /experiments/{expUUID}/outputs/{outputId}/configTypes/{typeId}
+    Returns a single configuration source type for given typeId: typeId, name, description, scope, expected parameter descriptors (e.g. "path" for file path)
 
-#### Sequence: Create configuration instance for an experiment
+    POST /experiments/{expUUID}/outputs/{outputId}
+    Create derived output. It returns a derived output descriptor with a unique Derived Output ID. It will contain the configuration used to create to allow manual reuse, as well as, the parent output ID. The parameter must include typeId and config adhering to the configuration type referenced by typeId.
```

Comment:
Are we providing an existing configuration instance id that includes parameters, or are we providing the explicit map of parameters according to the configuration source type? If the latter, would it be clearer to say 'must include typeId and parameters adhering...'?

Reply:
I think it's explicit map of parameters according to the sequence below? I was confused by the rich diff that shows the old deleted sequence instead of the new one... Thanks github.

Reply:
Updated as suggested